packages/ui5-tooling-modules/lib/utils/WebComponentRegistry.js

```diff
@@ -143,18 +144,24 @@ class RegistryEntry {
 				};
 			}
 
-			// UI5 normally accepts only one type for a property, except if "any" is used
-			// in this case we just use the first one as the primary type
+			// UI5 only accepts one type for a property/aggregation, in this case we just use the first one as the primary type.
 			parsedType = types[0];
 		}
 
 		// check if we have an array type
 		const arrayTypeMatch = parsedType?.match(/Array<(.*)>/i);
 		const multiple = !!arrayTypeMatch;
-		parsedType = arrayTypeMatch?.[1] || parsedType;
 
-		// complex types have a reference to other things, e.g. enums
-		if (typeInfo?.references) {
+		// [Complex types]:
+		// we have a reference to other things -> enums, interfaces, classes
+		if (typeInfo?.references?.length > 0) {
+			// Since the UI5 runtime only allows for 1 single type per property/aggregation, we take the first reference
+			parsedType = typeInfo.references[0].name;
```

Comment:
Ah, ok - this is interesting because the Luigi guys also used multiple types. So, the first type wins and should be then the most generic type - an alternative would be to use `object` in such cases.

Reply:
Yes, the runtime only evaluates one single type.
You can document more than one of course. To support multiple ones at runtime, you could also use `any` e.g. if you want to accept things like `object | string`.